tests/Unit/Domain/ShoppingListTest.php

```diff
@@ -54,11 +56,23 @@ public function testAddItem(): void
         $list = new ShoppingList(
             new Slug('my-groceries'),
             'My Groceries',
-            new ShoppingItemStack($item1),
+            false,
+            new ShoppingItemStack($item1)
         );
 
         $list->addItem($item2);
 
         $this->assertSame([$item1, $item2], $list->getItems()->all());
     }
+
+    /**
+     * @param ShoppingList $list
+     * @depends testConstructWithoutList
+     */
+    public function testSetArchived(ShoppingList $list): void
+    {
+        $list->setArchived(true);
```

Comment:
Required: once you've made the change to what the setter returns, this line needs to assert that the list is returned by the setter, i.e.:

```php
$this->assertSame($list, $list->setArchived(true));
```

You'll see the same pattern in the test for `setName()`.